src/affexpr.jl

```diff
@@ -15,23 +15,83 @@
 # Operator overloads in src/operators.jl
 #############################################################################
 
+# Utilities for OrderedDict
+function add_or_set!(dict::OrderedDict{K,V}, k::K, v::V) where {K,V}
+    # TODO: This unnecessarily requires two lookups for k.
+    # TODO: Decide if we want to drop zeros here.
```

Comment:
I am in favor of removing the zeros here

Reply:
This could have performance implications that I don't understand well. I'd want to check that the implementation handles deletion and re-adding efficiently or that benchmarks don't show a regression if we delete zeros.